builder/processor-tools/src/main/java/io/helidon/builder/processor/tools/BuilderTypeTools.java

```diff
@@ -97,48 +99,74 @@ public Optional<TypeInfo> createTypeInfo(AnnotationAndValue annotation,
+     * If it is not acceptable then the caller is expected to throw an exception or log an error, etc.
+     *
+     * @param element   the element
+     * @return true if the element is acceptable
+     */
+    protected boolean isAcceptableBuilderTarget(Element element) {
+        final ElementKind kind = element.getKind();
+        final Set<Modifier> modifiers = element.getModifiers();
+        boolean isAcceptable = (kind == ElementKind.INTERFACE
+                                        || kind == ElementKind.ANNOTATION_TYPE
+                                        || (ACCEPT_ABSTRACT_CLASS_TARGETS
+                                                    && (kind == ElementKind.CLASS && modifiers.contains(Modifier.ABSTRACT))));
+        return isAcceptable;
+    }
+
     /**
      * Translation the arguments to a collection of {@link io.helidon.pico.types.TypedElementName}'s.
      *
-     * @param element       the typed element (i.e., class)
-     * @param processingEnv the processing env
+     * @param element               the typed element (i.e., class)
+     * @param processingEnv         the processing env
+     * @param wantWhatWeCanAccept   pass true to get the elements we can accept to process, false for the other ones
      * @return the collection of typed elements
      */
-    protected Collection<TypedElementName> toElementInfo(TypeElement element, ProcessingEnvironment processingEnv) {
+    protected Collection<TypedElementName> toElementInfo(TypeElement element,
+                                                         ProcessingEnvironment processingEnv,
+                                                         boolean wantWhatWeCanAccept) {
         return element.getEnclosedElements().stream()
                 .filter(it -> it.getKind() == ElementKind.METHOD)
                 .map(ExecutableElement.class::cast)
-                .filter(this::canAccept)
+                .filter(it -> (wantWhatWeCanAccept == canAccept(it)))
                 .map(it -> createTypedElementNameFromElement(it, processingEnv.getElementUtils()))
                 .collect(Collectors.toList());
     }
 
     /**
-     * Returns true if the executable element passed is acceptable for processing (i.e., not a static and not a default method).
+     * Returns true if the executable element passed is acceptable for processing (i.e., not a static and not a default method
+     * on interfaces, and abstract methods on abstract classes).
      *
-     * @param ee the executable element
-     * @return true if not default and not static
+     * @param ee                the executable element
+     * @return true if not able to accept
      */
     protected boolean canAccept(ExecutableElement ee) {
         Set<Modifier> mods = ee.getModifiers();
-        return !mods.contains(Modifier.DEFAULT) && !mods.contains(Modifier.STATIC);
+        if (mods.contains(Modifier.ABSTRACT)) {
```

Comment:
what about `return mods.contains(Modifier.ABSTRACT)` instead of this?

Reply:
next commit